api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/ApplicationConfigurationOntologyUtils.java

```diff
@@ -18,112 +21,228 @@
+		for (ObjectProperty op : populatedObjectProperties) {
+		    fauxProperties.addAll(getPopulatedFauxObjectProperties(op, subject, vreq));
+		}
+		return fauxProperties;
+    }
+    
+    public static List<FauxDataPropertyWrapper> getPopulatedFauxDPs(List<DataProperty> populatedDataProperties, Individual subject, VitroRequest vreq) {
+		List<FauxDataPropertyWrapper> fauxProperties = new ArrayList<FauxDataPropertyWrapper>();
+		for (DataProperty dp : populatedDataProperties) {
+		    fauxProperties.addAll(getPopulatedFauxDataProperties(dp, subject, vreq));
+		}
+		return fauxProperties;
+    }
 
-        if(prop != null) {
-            log.debug("Checking " + prop.getURI() + " for additional properties");
-            queryStr = queryStr.replaceAll("For \\?property", "For <" + prop.getURI() + ">");
-        }
-        log.debug(queryStr);
-        Query q = QueryFactory.create(queryStr);
+	private static List<FauxObjectPropertyWrapper> getPopulatedFauxObjectProperties(ObjectProperty op, Individual subject, VitroRequest vreq) {
+		Model displayModel = vreq.getDisplayModel();
+        Model tboxModel = vreq.getOntModelSelector().getTBoxModel();
+		Model union = ModelFactory.createUnion(displayModel, tboxModel);
+		WebappDaoFactory wadf = new WebappDaoFactoryJena(ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM, union));
+		FauxPropertyDao fpDao = wadf.getFauxPropertyDao();
+        
+	    Query q = createQuery(op, false);
         QueryExecution qe = QueryExecutionFactory.create(q, union);
-        WebappDaoFactory wadf = new WebappDaoFactoryJena(
-                ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM, union));
-        ObjectPropertyDao opDao = wadf.getObjectPropertyDao();
+	    List<FauxObjectPropertyWrapper> fauxObjectProps = new ArrayList<FauxObjectPropertyWrapper>();
         try {
             ResultSet rs = qe.execSelect();
             while (rs.hasNext()) {
                 QuerySolution qsoln = rs.nextSolution();
                 log.debug(qsoln);
-                String opURI = (prop != null) ? prop.getURI() : qsoln.getResource(
-                        "property").getURI();
                 Resource domainRes = qsoln.getResource("domain");
                 String domainURI = (domainRes != null) ? domainRes.getURI() : null;
-                String rangeURI = qsoln.getResource("range").getURI();
-                if (appropriateDomain(domainRes, subject, tboxModel)) {
-                    ObjectProperty faux = opDao.getObjectPropertyByURIs(
-                            opURI, domainURI, rangeURI, (prop != null) ? prop.clone() : null);
-                    if (faux != null) {
-                        additionalProps.add(faux);
-                    } else {
-                        log.error("Could not retrieve " + opURI + " qualified by " +
-                                " domain " + domainURI + " and range " + rangeURI);
-                    }
+                //String rangeURI = qsoln.getResource("range").getURI();
```

Comment:
Why would you like to keep this line in comment? Do you have some scenario on top of your mind in which we should uncomment this?

Reply:
No need to keep it. Removed.